tests/bsim/bluetooth/mesh/src/test_blob.c

```diff
@@ -1593,6 +1594,89 @@ static void test_srv_stop(void)
 }
 #endif /* CONFIG_BT_SETTINGS */
 
+static void test_cli_friend_pull(void)
```

Comment:
I wonder if we could re-use `test_cli_trans_persistency_pull`. For example, semaphores initialization and device setup (except `bt_mesh_test_cfg_set`) could be re-used. Then test both add as much targets as the want. Then the rest code with `blob_cli_xfer` initialization and transfer could be re-used as well.

Same with the server.

Reply:
I'll abstract what I can into common function.
>Then both test add as much targets as they want
This part can't be common for both I think. I don't want to introduce new test parameters when I can do without it, so I don't want to create variable to store how many targets will there be (like `dfu_targets_cnt`). I think sending BLOB to limit of `CONFIG_BT_MESH_FRIEND_LPN_COUNT` is ok. We need to know how many targets there are, so we can wait for friendship beeing established for every one of them.

I'll create common function that setups device, provisions, inits semaphores and configures xfer.

For server I think I can share device setup and semaphore inits - friendship must be established before `bt_mesh_blob_srv_recv` is called and no failures are expected.